com/williamfiset/algorithms/geometry/ConvexPolygonCutWithLineSegment.java

```diff
@@ -64,8 +64,8 @@ private static int orientation(double ax, double ay, double bx, double by, doubl
 
   // Example usage
   public static void main(String[] args) {
-
-    Pt[] squarePolygon = {new Pt(0, 0), new Pt(0, 4), new Pt(4, 0), new Pt(4, 4)};
+    //Input points should be ordered in terms of the geometrical shape of the polygon.
```

Comment:
I would put this comment near the cut method so we know what the input should look like.

I think more accurately you want to say that the points should be sorted in CW or CCW order, is that correct? If so ould you mind creating a method that sorts points in CW/CCW order? You can re-use the PointComparator logic in the convex hull algorithm:

https://github.com/williamfiset/Algorithms/blob/master/com/williamfiset/algorithms/geometry/ConvexHullMonotoneChainsAlgorithm.java

Reply:
Actually, I'm not so sure sorting the points would work because we want to retain the original shape of the polygon. Feel free to just move the comment that warms about the point ordering.

Reply:
Okay, I've made the changes. But I think as the polygon is convex, sorting the vertices in the CW/CCW order would work fine.
Because I think the original shape of a convex polygon will be the same as CW/CCW order.

Reply:
Okay Hello @williamfiset  
I have written that function and everything works fine with that. Should I commit the new code?

Reply:
Hello @williamfiset 
Turned out that sorting points in the CW direction works fine. 
I've made the function and committed the changes.
Please review the code.